platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/stale/StaleEventDetectorMetrics.java

```diff
@@ -70,7 +71,9 @@ public void reportStaleEvent(@NonNull final GossipEvent event) {
         int systemTransactions = 0;
         int appTransactions = 0;
 
-        for (final ConsensusTransactionImpl transaction : event.getHashedData().getTransactions()) {
+        final Iterator<Transaction> iterator = event.transactionIterator();
```

Comment:
Why do you switched to exliziet iterator usage instead of forEach?

Reply:
the event interface only provides and `Iterator`, not an `Iterable`, so I cannot use forEach. This will change once transactions are no longer stored in arrays